code/hsec-tools/app/Main.hs

```diff
@@ -31,12 +33,31 @@ cliOpts = info (commandsParser <**> helper) (fullDesc <> header "Haskell Advisor
     commandsParser =
       subparser
         (  command "check" (info commandCheck (progDesc "Syntax check a single advisory"))
+        <> command "reserve-id" (info commandReserve (progDesc "Reserve an HSEC ID"))
         <> command "osv" (info commandOsv (progDesc "Convert a single advisory to OSV"))
         <> command "render" (info commandRender (progDesc "Render a single advisory as HTML"))
         <> command "generate-index" (info commandGenerateIndex (progDesc "Generate an HTML index"))
         <> command "help" (info commandHelp (progDesc "Show command help"))
         )
 
+commandReserve :: Parser (IO ())
+commandReserve =
+  Command.Reserve.go
+  <$> optional (argument str (metavar "REPO"))
+  <*> flag
+        Command.Reserve.DoNotAssignId -- default value
```

Comment:
Perhaps `assign` could be the default?

Reply:
I'm not sure to see the use case to not have it enabled, can you give an example please?

Reply:
Eventually, we want people to submit advisories (and reservations) as `HSEC-0000-0000.md`, and have the PR automation assign the IDs.  That is why `--assign` is not the default.

Reply:
I would instead propose that the contributor guide, when updated to mention this tool, recommends to use `--assign`.  Then, when the PR automation we desire has been established, we can remove the flag from the contributor how-to doc.

Reply:
I think it'll make more sense in another command, a `reserve-id` with "no ID" by default does not seem natural to me.

Reply:
The process we are (eventually) aiming for is that contributor PRs would use placeholder IDs, and the automation will take care of assigning the appropriate IDs.  In this workflow the preferred user action is to use `hsec-tools` to create the placeholder reservations.  (Similarly with actual advisories - they will submit PR using placeholder IDs, and the automation will assign appropriate IDs).

So, maybe we just need a better subcommand name or CLI options.  For example, instead of *any* default we could require user to specify one of `--placeholder` or `--assign`  Then, as our tooling and automation evolve, we only need to update the documentation as to which option users should use in the typical workflow.

Reply:
It sounds better.

Maybe:

```
new --placeholder
new --reserved
```

I not sure what would be better `new`, `new-advisory`, `create`, `create-advisory`, any opinion on that?

Reply:
I thought about it some more.  Instead of multiple flags, we could have a single option:

```
hsec-tools reserve --id-mode={placeholder|auto}
```

For now there would be no default, meaning user must specify one of the values.  Later when the workflow has the desired automations, we might make `placeholder` the default.

Very much open to other ideas for the command, option and option value names.

Reply:
As for subcommand names, I like `new`.

Reply:
I like this design

Reply:
PR updated with this design.